pkg/image/api/helper.go

```diff
@@ -157,12 +157,28 @@ func (r DockerImageReference) String() string {
 		registry += "/"
 	}
 
-	if len(r.Namespace) == 0 {
-		r.Namespace = DockerDefaultNamespace
+	namespace := r.Namespace
+	if len(namespace) == 0 {
+		namespace = DockerDefaultNamespace
+	}
+	namespace += "/"
+
+	return fmt.Sprintf("%s%s%s", registry, namespace, r.NameString())
+}
+
+// StringNoTag converts a DockerImageReference to a Docker pull spec with no tag.
+func (r DockerImageReference) StringNoTag() string {
```

Comment:
This also needs test cases - please open a new PR that undoes these changes with the requested changes described above.